### PR TITLE
deps!: Update min Flutter to v3.13.0, compatibility with v3.16.8

### PR DESCRIPTION
**Pull request by Gustl22**
# Description

Update minimum required Flutter version to 3.13.0 as Linux requires at least that version to work properly due to a bug.
Therefore we also only test against 3.13.0 and higher (here 3.16.1).
Also remove `dartdoc` dependency introduced in 7ca5058acbaab061dfcc18c61fbdf42b7fc5ed87 as the package is integrated in dart SDK since v5.0.0.

Fix pump Linux, introduced in https://github.com/flutter/flutter/pull/135266 (https://github.com/flutter/flutter/commit/5def6f2bab65ab05f772a8e55ec2fff71822090d)

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `refactor:`,
      `docs:`, `chore:`, `test:`, `ci:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation and added dartdoc comments with `///`, where necessary.
- [x] I have updated/added relevant examples in [example].

## Breaking Change

- [x] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

### Migration instructions

```shell
flutter upgrade
```

## Related Issues

[issue database]: https://github.com/bluefireteam/audioplayers/issues
[Contributor Guide]: https://github.com/bluefireteam/audioplayers/blob/main/contributing.md#feature-requests--prs
[Conventional Commit]: https://conventionalcommits.org
[example]: https://github.com/bluefireteam/audioplayers/tree/main/packages/audioplayers/example

### CODE REVIEW

File: .github/workflows/build-example.yml [hunk 1]
```diff
@@ -9,11 +9,9 @@ on:
         type: choice
         options:
           - 'any'
+          - '3.16.x'
           - '3.13.x'
           - '3.10.x'
```
Comment (spydon) [inline .github/workflows/build-example.yml:14]: Shouldn't 3.10.x be removed too?

**Review by spydon — APPROVED** [review]

**Comment by spydon** [comment]
This is a `deps!:` btw, since it is not a new feature

File: .github/workflows/build-example.yml (hunk 1 above)
Comment (Gustl22) [reply .github/workflows/build-example.yml:14]: We can remove it, I just left it there to be able to test against unsupported flutter versions. Don't know if that makes sense, but we can remove it, to not spread confusion.

**Comment by Gustl22** [comment]
> This is a deps!: btw, since it is not a new feature

Thank you, wasn't aware of that tag.

**Review by spydon — APPROVED** [review]

**Review by spydon — APPROVED** [review]
Lgtm!

**Comment by Gustl22** [comment]
Blocked by #1732

**Review by spydon — APPROVED** [review]

**Pull request merged**